### PR TITLE
PLAT-419: don't update objest.State if pulse was changed

### PR DESCRIPTION
**Pull request by ivanshibitov**
**- What I did**
Prevent updating of object shared state if pulse was changed

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/virtual/execute/execute.go [hunk 1]
```diff
@@ -543,16 +543,22 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
...
 		_, class, memory = executionNewState.Amend()
-		action = s.setNewState(class, memory)
+		action = s.getUpdateStateAction(class, memory)
 	case requestresult.SideEffectDeactivate:
 		panic(throw.NotImplemented())
 	default:
 		panic(throw.IllegalValue())
 	}
 
+	if s.migrationHappened {
```
Comment (sexyhamster) [inline ledger-core/virtual/execute/execute.go:556]: this should be done only if requestresult.SideEffectActivate or requestresult.SideEffectAmend
it is possible that nothing should be done with descriptor

Comment (sexyhamster) [reply ledger-core/virtual/execute/execute.go:556]: and test for this case should exist >_<

Comment (ivanshibitov) [reply ledger-core/virtual/execute/execute.go:556]: "it is possible that nothing should be done with descriptor" - What do you mean? I shouldn't change s.execution.ObjectDescriptor inside 'if' statement?

Comment (sexyhamster) [reply ledger-core/virtual/execute/execute.go:556]: My point is that we should not send latest state if state hasn't changed.

**Pull request merged**